### PR TITLE
Encryption: Extract encryption into service

### PR DESCRIPTION
**Pull request by undef1nd**
**What this PR does / why we need it**:
In order to support envelope encryption, `util.Encrypt` and `util.Decrypt` need to be extracted into a service rather than stay as global functions.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/encryption/ossencryption/ossencryption_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		encrypted, err := svc.Encrypt([]byte("grafana"), "1234")
+		require.NoError(t, err)
+
+		decrypted, err := svc.Decrypt(encrypted, "1234")
+		require.NoError(t, err)
+
+		assert.Equal(t, []byte("grafana"), decrypted)
+	})
+
+	t.Run("decrypting empty payload should not fail", func(t *testing.T) {
```
Comment (sakjur) [inline pkg/services/encryption/ossencryption/ossencryption_test.go:33]:
```suggestion
	t.Run("decrypting empty payload should return error", func(t *testing.T) {
```

**Review by sakjur — COMMENTED** [review]
This looks promising, but let's wait until Wire's been merged in #32289 — it should happen today.

**Comment by joanlopez** [comment]
> This looks promising, but let's wait until Wire's been merged in #32289 — it should happen today.

Yeah, that's completely fair. 

In fact, we already discussed it with @undef1nd that would be better to wait til Wire's get merged.

Thanks!

File: pkg/services/encryption/ossencryption/ossencryption.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package ossencryption
```
Comment (joanlopez) [inline pkg/services/encryption/ossencryption/ossencryption.go:1]: I'm wondering if we should rename to something else like just `service` or `encryptionservice` cause `oss` sounds redundant within an open source code base 🤭  Though we already have some "OSS" references as part of other packages names / types.

What do you think? @sakjur 

Thanks!! :)

**Comment by joanlopez** [comment]
I have updated it to migrate it to Wire, so I guess it's ready again to be reviewed :) Thanks!!

**Comment by joanlopez** [comment]
> This looks promising, but let's wait until Wire's been merged in #32289 — it should happen today.

Hey @sakjur, do you think we can merge it? 🤔  Thanks!

File: pkg/services/encryption/ossencryption/ossencryption.go (hunk 2 above)
Comment (sakjur) [reply pkg/services/encryption/ossencryption/ossencryption.go:1]: I'm not fond of `OSS` in package names either and try to not use it myself, but I don't have a better alternative in mind.

But yeah, we've done it elsewhere 🤷

**Review by sakjur — APPROVED** [review]

**Pull request merged**